### PR TITLE
NXP-25325: Submit Search Form when enter is pressed on full text search

### PR DESCRIPTION
**Pull request by nmpcunha**

### CODE REVIEW

File: elements/search/default/nuxeo-default-search-form.html [hunk 1]
```diff
@@ -136,7 +136,10 @@
         searchTerm: {
           type: String,
           notify: true
-        }
+        },
+        auto: {
+          type: Boolean
+        },
```
Comment (guirenard) [inline elements/search/default/nuxeo-default-search-form.html]: not sure lint won't complain about the trailing comma

**Comment by guirenard** [comment]
We could actually add a ftest on this!

File: elements/search/nuxeo-search-form.html [hunk 2]
```diff
@@ -827,6 +827,16 @@ <h2>[[i18n('searchForm.delete.heading')]]</h2>
           }
         }.bind(this));
         this.form.addEventListener('trigger-search', this._search.bind(this));
+        if(!this.auto){
```
Comment (afjusto) [inline elements/search/nuxeo-search-form.html]: format

File: elements/search/nuxeo-search-form.html [hunk 3]
```diff
@@ -827,6 +827,16 @@ <h2>[[i18n('searchForm.delete.heading')]]</h2>
...
+          this.form.addEventListener("keypress", this._keyPressedListener.bind(this));
+        }
+      },
+
+      _keyPressedListener: function(e) {
+        // When Enter is pressed on an input element, the search should be triggered.
+        if (e.keyCode === 13 && e.composedPath()[0].tagName.toLowerCase() === 'input'){
```
Comment (afjusto) [inline elements/search/nuxeo-search-form.html]: format 😬

**Comment by afjusto** [comment]
👍

**Comment by guirenard** [comment]
Aren't you sure it is not going to trigger the search twice in case the search is `auto`?

**Comment by guirenard** [comment]
:+1:

**Comment by Gabez0r** [comment]
👍

**Pull request merged**